staging/src/k8s.io/legacy-cloud-providers/azure/retry/azure_error.go

```diff
@@ -224,6 +236,23 @@ func getRetryAfter(resp *http.Response) time.Duration {
 	return dur
 }
 
+// GetErrorWithRetriableHTTPStatusCodes gets an error with RetriableHTTPStatusCodes.
+// It is used to retry on some HTTPStatusCodes.
+func GetErrorWithRetriableHTTPStatusCodes(resp *http.Response, err error, retriableHTTPStatusCodes []int) *Error {
+	rerr := GetError(resp, err)
+	if rerr == nil {
+		return nil
+	}
+
+	for _, code := range retriableHTTPStatusCodes {
+		if rerr.HTTPStatusCode == code {
+			rerr.Retriable = true
```

Comment:
Can we break here if the `code` matches and not run the entire loop after match?

Reply:
good catch. fixed now